src/adapter_test_helpers/sumo_basic_test_helper.erl

```diff
@@ -176,6 +177,24 @@ check_proper_dates(Config) ->
   Date = Module:birthdate(Person),
   ok.
 
+-spec persist_using_changeset(config()) -> ok.
+persist_using_changeset(Config) ->
+  {_, Name} = lists:keyfind(name, 1, Config),
+  Module = sumo_config:get_prop_value(Name, module),
+  Schema = Module:sumo_schema(),
+
+  [] = sumo:find_by(Name, [{name, <<"John">>}]),
+  [P1] = sumo:find_by(Name, [{name, <<"A">>}]),
+  Doc1 = sumo:new_doc(people, P1),
```

Comment:
I still don't like sharing our `sumo_internal` types with the regular `sumo` user.